### PR TITLE
Add test that kills Focus app from recent apps, and restarts

### PR DESCRIPTION
**Pull request by npark-mozilla**

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/focus/activity/TestHelper.java [hunk 1]
```diff
@@ -175,9 +176,8 @@ static void pressEnterKey() {
     static void pressBackKey() {
         mDevice.pressBack();
     }
-    static void pressHomeKey() {
-        mDevice.pressHome();
-    }
+    static void pressHomeKey() { mDevice.pressHome(); }
```
Comment (sputnikci) [inline app/src/androidTest/java/org/mozilla/focus/activity/TestHelper.java]: \[Checkstyle\] ERROR: '\{' at column 32 should have line break after\.

File: app/src/androidTest/java/org/mozilla/focus/activity/TestHelper.java [hunk 2]
```diff
@@ -175,9 +176,8 @@ static void pressEnterKey() {
...
+    static void pressRecentAppsKey() throws RemoteException { mDevice.pressRecentApps(); }
```
Comment (sputnikci) [inline app/src/androidTest/java/org/mozilla/focus/activity/TestHelper.java]: \[Checkstyle\] ERROR: '\{' at column 61 should have line break after\.

File: app/src/androidTest/java/org/mozilla/focus/activity/RestartTest.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+import static junit.framework.Assert.assertTrue;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.mozilla.focus.activity.TestHelper.waitingTime;
+import static org.mozilla.focus.fragment.FirstrunFragment.FIRSTRUN_PREF;
+
+// This test opens enters and invalid URL, and Focus should provide an appropriate error message
```
Comment (pocmo) [inline app/src/androidTest/java/org/mozilla/focus/activity/RestartTest.java]: I guess this comment is not right?

File: app/src/androidTest/java/org/mozilla/focus/activity/RestartTest.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+                    .edit()
+                    .putBoolean(FIRSTRUN_PREF, true)
+                    .apply();
+
+            webServer = new MockWebServer();
+
+            try {
+                webServer.enqueue(new MockResponse()
+                        .setBody(TestHelper.readTestAsset("image_test.html"))
+                        .addHeader("Set-Cookie", "sphere=battery; Expires=Wed, 21 Oct 2035 07:28:00 GMT;"));
```
Comment (pocmo) [inline app/src/androidTest/java/org/mozilla/focus/activity/RestartTest.java]: Does Cookie headers from the other test shouldn't be needed here?

File: app/src/androidTest/java/org/mozilla/focus/activity/RestartTest.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+                webServer.close();
+                webServer.shutdown();
+            } catch (IOException e) {
+                throw new AssertionError("Could not stop web server", e);
+            }
+        }
+    };
+
+    @Test
+    public void settingsToFocus() throws InterruptedException, UiObjectNotFoundException, RemoteException {
```
Comment (pocmo) [inline app/src/androidTest/java/org/mozilla/focus/activity/RestartTest.java]: settingsToFocus?

**Pull request merged**